app/src/main/java/org/p2p/wallet/rpc/repository/history/RpcHistoryRemoteRepository.kt

```diff
@@ -0,0 +1,89 @@
+import org.p2p.solanaj.model.types.Encoding
+import org.p2p.solanaj.model.types.RequestConfiguration
+import org.p2p.solanaj.model.types.RpcRequest
+import org.p2p.solanaj.utils.crypto.Base64Utils
+import org.p2p.wallet.rpc.api.RpcHistoryApi
+
+class RpcHistoryRemoteRepository(
+    private val rpcApi: RpcHistoryApi
+) : RpcHistoryRepository {
+
+    override suspend fun sendTransaction(transaction: Transaction): String {
+        val serializedTransaction = transaction.serialize()
+
+        val base64Trx = Base64Utils.encode(serializedTransaction)
+            .replace("\n", "")
+
+        val params = mutableListOf<Any>()
+
+        params.add(base64Trx)
+        params.add(RequestConfiguration(encoding = Encoding.BASE64.encoding))
+
+        val rpcRequest = RpcRequest("sendTransaction", params)
+        return rpcApi.sendTransaction(rpcRequest).result
+    }
+
+    override suspend fun simulateTransaction(transaction: Transaction): String {
+        val serializedTransaction = transaction.serialize()
+
+        val base64Trx = Base64Utils.encode(serializedTransaction)
+            .replace("\n", "")
+
+        val params = mutableListOf<Any>()
+
+        params.add(base64Trx)
+        params.add(RequestConfiguration(encoding = Encoding.BASE64.encoding))
+
+        val rpcRequest = RpcRequest("simulateTransaction", params)
+        val result = rpcApi.simulateTransaction(rpcRequest).result
+        if (result.value.error != null) {
+            throw IllegalStateException("Transaction simulation failed: ${result.value.linedLogs()}")
+        } else {
+            return ""
+        }
+    }
+
+    override suspend fun sendTransaction(serializedTransaction: String): String {
+        val base64Trx = serializedTransaction.replace("\n", "")
+
+        val params = mutableListOf<Any>()
+        params.add(base64Trx)
+        params.add(RequestConfiguration(encoding = Encoding.BASE64.encoding))
+
+        val rpcRequest = RpcRequest("sendTransaction", params)
+        return rpcApi.sendTransaction(rpcRequest).result
+    }
+
+    override suspend fun simulateTransaction(serializedTransaction: String): String {
+        val base64Trx = serializedTransaction.replace("\n", "")
+
+        val params = mutableListOf<Any>()
+        params.add(base64Trx)
+        params.add(RequestConfiguration(encoding = Encoding.BASE64.encoding))
+
+        val rpcRequest = RpcRequest("simulateTransaction", params)
+        val result = rpcApi.simulateTransaction(rpcRequest).result
+        if (result.value.error != null) {
+            throw IllegalStateException("Transaction simulation failed: ${result.value.linedLogs()}")
+        } else {
+            return ""
+        }
+    }
+
+    override suspend fun getConfirmedTransactions(
+        signatures: List<String>
+    ): List<ConfirmedTransactionParsed> {
+        val requestsBatch = signatures.map {
+            val encoding = mapOf("encoding" to "jsonParsed")
```

Comment:
Strings `encoding` and `jsonParsed` are used widely. Please consider to move them out as constants

Reply:
I think we need discuss about this, cause this values used also in Solana module, so before refactor this i think we need determine how we will solve this